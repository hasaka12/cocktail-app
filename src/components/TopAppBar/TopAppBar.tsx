import AppBar from "@mui/material/AppBar";
import Box from "@mui/material/Box";
import Toolbar from "@mui/material/Toolbar";
import Typography from "@mui/material/Typography";
import Button from "@mui/material/Button";

import { NavigationService } from "../../services";
import { Route } from "../../constants/routes-enum";
import { RootState } from "../../store/store";
import { useSelector } from "react-redux";

interface NavItem {
  name: string;
  route: Route;
}
const TopAppBar = () => {
  const navItems: Array<NavItem> = [
    { name: "Search", route: Route.Search },
    { name: "Favorites", route: Route.Favorite },
  ];
  const favoriteCocktail = useSelector(
    (state: RootState) => state.favoriteCocktail.data
  );

  console.log(favoriteCocktail.length);

  return (
    <Box sx={{ flexGrow: 1, mb: 2 }}>
      <AppBar position="sticky">
        <Toolbar>
          <Typography
            variant="h6"
            noWrap
            component="div"
            style={{ cursor: "pointer" }}
            sx={{ flexGrow: 1 }}
            onClick={() => {
              NavigationService.navigate(Route.Home);
            }}
          >
            Cocktail Maker
          </Typography>
          <Box>
            {navItems.map((item: NavItem) => (
              <Button
                key={item.name}
                sx={{ color: "#fff" }}
                onClick={() => {
                  NavigationService.navigate(item.route);
                }}
              >
                {`${item.name} ${
                  item.name === "Favorites" &&
                  favoriteCocktail &&
                  favoriteCocktail.length
                    ? `(${favoriteCocktail.length})`
                    : ""
                }`}
              </Button>
            ))}
          </Box>
        </Toolbar>
      </AppBar>
    </Box>
  );
};

export default TopAppBar;
