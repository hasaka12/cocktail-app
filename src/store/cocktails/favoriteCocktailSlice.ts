import { PayloadAction, createSlice } from "@reduxjs/toolkit";
import { CocktailsDto } from "../../models";

interface CounterState {
  data: CocktailsDto[];
}

const initialState: CounterState = {
  data: [],
};

const counterSlice = createSlice({
  name: "my-counter",
  initialState,
  reducers: {
    addFavoriteCocktails: (state, action: PayloadAction<CocktailsDto>) => {
      if (state.data.some((item) => item.idDrink === action.payload.idDrink)) {
        const index = state.data.findIndex(
          (item) => item.idDrink === action.payload.idDrink
        );
        state.data.splice(index, 1);
        return;
      }
      state.data.push(action.payload);
    },
  },
});

export const { addFavoriteCocktails } = counterSlice.actions;

export default counterSlice.reducer;
