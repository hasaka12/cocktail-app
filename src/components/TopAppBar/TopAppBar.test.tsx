import { render, screen } from "@testing-library/react";
import { Store, UnknownAction, configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";

import favoriteCocktailSlice from "../../store/cocktails/favoriteCocktailSlice";

import TopAppBar from "./TopAppBar";

describe("TopAppBar", () => {
  let store: Store<unknown, UnknownAction, unknown>;

  beforeEach(() => {
    store = configureStore({
      reducer: {
        favoriteCocktail: favoriteCocktailSlice,
      },
    });
  });

  test("renders app bar with navigation items", () => {
    render(
      <Provider store={store}>
        <TopAppBar />
      </Provider>
    );

    const searchButton = screen.getByText("Search");
    const favoritesButton = screen.getByText("Favorites");
    expect(searchButton).toBeInTheDocument();
    expect(favoritesButton).toBeInTheDocument();
  });

  test("Renders correctly", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <TopAppBar />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
