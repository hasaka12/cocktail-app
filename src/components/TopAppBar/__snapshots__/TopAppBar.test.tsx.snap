// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopAppBar Renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-dc7zdy"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionSticky css-7e0s21-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-1dvstpp-MuiTypography-root"
          style="cursor: pointer;"
        >
          Cocktail Maker
        </div>
        <div
          class="MuiBox-root css-0"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1kkpk3h-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Search 
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1kkpk3h-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Favorites 
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
